lib/membrane_opus/util.ex

```diff
@@ -69,17 +70,27 @@ defmodule Membrane.Opus.Util do
 
   @spec validate_pts_integrity([Membrane.Buffer.t()], integer()) :: :ok
   def validate_pts_integrity(packets, input_pts) do
-    cond do
-      length(packets) < 2 or Enum.at(packets, 1).pts == input_pts ->
-        :ok
+    %Buffer{pts: output_pts} = List.first(packets)
 
-      Enum.at(packets, 1).pts > input_pts ->
-        Membrane.Logger.warning("PTS values are overlapping")
-        :ok
+    if output_pts == nil or input_pts == nil do
+      :ok
+    else
+      # Diff constantly oscillates between approximately 2 ms - 22 ms.
+      diff = output_pts - input_pts
+      epsilon = 30 |> milliseconds()
```

Comment:
As we spoke, the actual overlap is no more than 2ms. Try refactoring it so that we operate on the actual overlap.